src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2060,6 +2059,62 @@ UpdateShardPlacementState(uint64 placementId, char shardState)
 }
 
 
+/*
+ * UpdatePlacementGroupId sets the groupId for the placement identified
+ * by placementId.
+ */
+void
+UpdatePlacementGroupId(uint64 placementId, int groupId)
+{
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 1;
+	bool indexOK = true;
+	Datum values[Natts_pg_dist_placement];
+	bool isnull[Natts_pg_dist_placement];
+	bool replace[Natts_pg_dist_placement];
+	bool colIsNull = false;
+
+	Relation pgDistPlacement = table_open(DistPlacementRelationId(), RowExclusiveLock);
+	TupleDesc tupleDescriptor = RelationGetDescr(pgDistPlacement);
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_placement_placementid,
+				BTEqualStrategyNumber, F_INT8EQ, Int64GetDatum(placementId));
+
+	SysScanDesc scanDescriptor = systable_beginscan(pgDistPlacement,
+													DistPlacementPlacementidIndexId(),
+													indexOK,
+													NULL, scanKeyCount, scanKey);
+
+	HeapTuple heapTuple = systable_getnext(scanDescriptor);
+	if (!HeapTupleIsValid(heapTuple))
```

Comment:
Well, this is not possible to trigger, but we'd like to have these checks in general for corrupted metadata. Few more such checks better to stay?

Reply:
Yeah, makes sense.